h/services/annotation_write.py

```diff
@@ -221,7 +221,10 @@ def _validate_group(self, annotation: Annotation, enforce_write_permission=True)
     def upsert_annotation_slim(self, annotation):
         self._db.flush()  # See the last model changes in the transaction
 
-        if not annotation.group:
+        user_id = self._db.scalar(
```

Comment:
Moved this to the top to handle both cases, group and user, in the same conditional.